utils/auth.js

```diff
@@ -0,0 +1,9 @@
+const withAuth = (req, res, next) => {
+    if (!req.session.user_id) {
```

Comment:
Isn't user_id set when the session is created so this will not check the authentication status of the user but will check to see if they have a session.